pkg/resources/factory.go

```diff
@@ -119,3 +120,14 @@ func (rf *resourceFactory) GetResourcePool(rc *types.ResourceConfig, deviceList
 	rPool := newResourcePool(rc, apiDevices, devicePool)
 	return rPool, nil
 }
+
+// GetRdmaSpec returns the vendors rdma specs
+func (rf *resourceFactory) GetRdmaSpec(vendor, pciAddrs string) types.RdmaSpec {
+	switch vendor {
+	// Mellanox device
+	case "15b3":
```

Comment:
Does this have to be vendor specific creation or there can be a general way to discovering rdma capability of a network pci device as a kernel device?

Reply:
Vendors will mostly have only one approach to support rdma and have there own resources paths to mount

Reply:
I was thinking if we can use a generic filesystem path to discover whether a kernel device is capable of doing rdma, for example, read a certain path under ``/sys/bus/pci/<id>`` to determine if that pci device supports rdma. Is this part vendor specific?